graphql/client.go

```diff
@@ -83,6 +123,24 @@ func NewClientUsingGet(endpoint string, httpClient Doer) Client {
 	return newClient(endpoint, httpClient, http.MethodGet)
 }
 
+// NewClientUsingWebSocket returns a [WebSocketClient] which makes subscription requests
+// to the given endpoint using webSocket.
+//
+// The client does not support queries nor mutations, and will return an error
+// if passed a request that attempts one.
+func NewClientUsingWebSocket(endpoint string, wsDialer Dialer, headers http.Header) WebSocketClient {
+	if headers == nil {
+		headers = http.Header{}
+	}
+	headers.Add("Sec-WebSocket-Protocol", "graphql-transport-ws")
```

Comment:
Do you need to force `graphql-transport-ws` sub-protocol? We are using `graphql-ws` in our case. Had to override this header in `DialContext` to make it work.

Reply:
change made [here](https://github.com/Khan/genqlient/pull/294#discussion_r1348374734)